openl3/cli.py

```diff
@@ -82,23 +88,59 @@ def run(inputs, output_dir=None, suffix=None, input_repr="mel256", content_type=
+        print('openl3: No files found in {}. Aborting.'.format(str(inputs)))
         sys.exit(-1)
 
     # Load model
-    model = load_embedding_model(input_repr, content_type, embedding_size)
-
-    # Process all files in the arguments
-    for filepath in file_list:
-        if verbose:
-            print('openl3: Processing: {}'.format(filepath))
-        process_file(filepath,
-                     output_dir=output_dir,
-                     suffix=suffix,
-                     model=model,
-                     center=center,
-                     hop_size=hop_size,
-                     verbose=verbose)
+    if modality == 'audio':
+        model = load_audio_embedding_model(input_repr, content_type,
+                                           audio_embedding_size)
+
+        # Process all files in the arguments
+        for filepath in file_list:
+            if verbose:
+                print('openl3: Processing: {}'.format(filepath))
+            process_audio_file(filepath,
+                               output_dir=output_dir,
+                               suffix=suffix,
+                               model=model,
+                               center=audio_center,
+                               hop_size=audio_hop_size,
+                               verbose=verbose)
+    elif modality == 'image':
+        model = load_image_embedding_model(input_repr, content_type,
+                                           image_embedding_size)
+
+        # Process all files in the arguments
+        for filepath in file_list:
+            if verbose:
+                print('openl3: Processing: {}'.format(filepath))
+            process_image_file(filepath,
+                               output_dir=output_dir,
+                               suffix=suffix,
+                               model=model,
+                               verbose=verbose)
+    elif modality == 'video':
+        audio_model = load_audio_embedding_model(input_repr, content_type,
+                                                 audio_embedding_size)
+        image_model = load_image_embedding_model(input_repr, content_type,
+                                                 image_embedding_size)
+
+        # Process all files in the arguments
+        for filepath in file_list:
+            if verbose:
+                print('openl3: Processing: {}'.format(filepath))
+            process_video_file(filepath,
+                               output_dir=output_dir,
+                               suffix=suffix,
+                               audio_model=audio_model,
+                               image_model=image_model,
+                               audio_embedding_size=audio_embedding_size,
+                               audio_center=audio_center,
+                               audio_hop_size=audio_hop_size,
+                               image_embedding_size=image_embedding_size,
+                               verbose=verbose)
+
```

Comment:
Ah, I was assuming that this would only be run by the CLI, in which case the choice of the `modality` string would be limited to valid choices. But it wouldn't hurt to add one here in case someone decides to run this as a standalone function.

Reply:
yes, can't hurt